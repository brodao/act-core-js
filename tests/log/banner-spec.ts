import ACT from '../../lib/index';

const { wrap } = require('@brodao/act-jest-snapshot-console');

describe("Testa a apresentação ou não do 'splash' completo", () => {
	test("Apresenta o 'splash'", () => {
		ACT.logger._config.showBanner = true;

		expect(
			wrap(() =>
				ACT.logger.showBanner({
					name: 'test_show_banner',
					version: '99.99.99',
					description: 'Show Banner',
				})
			)
		).toMatchSnapshot();
	});

	test("Apresenta um 'splash' simples de identificação", () => {
		ACT.logger._config.showBanner = false;

		expect(
			wrap(() =>
				ACT.logger.showBanner({
					name: 'test_show_banner',
					version: '99.99.99',
					description: 'Show Banner',
				})
			)
		).toMatchSnapshot();
	});
});
