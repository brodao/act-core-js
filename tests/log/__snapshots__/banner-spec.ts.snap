// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa a apresentação ou não do 'splash' completo Apresenta o 'splash' 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "",
}
`;

exports[`Testa a apresentação ou não do 'splash' completo Apresenta um 'splash' simples de identificação 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "",
}
`;
